src/dmd/backend/cg87.d

```diff
@@ -3639,7 +3639,7 @@ void fixresult_complex87(ref CodeBuilder cdb,elem *e,regm_t retregs,regm_t *pret
     tym_t tym = tybasic(e.Ety);
     uint sz = _tysize[tym];
 
-    if (*pretregs == 0 && retregs == mST01)
+    if ((*pretregs == 0 || *pretregs & mCX) && retregs == mST01)
```

Comment:
I don't think this is right, I think I need to generate code to mov the value from rtregs to pretregs (I can probably look at fixresult87 for an example of how to do this).